### PR TITLE
Clarifying the Final label in Inspect submissions view

### PR DESCRIPTION
**Pull request by PasiSa**
The Final label is used in Inspect submissions' submission list
also for submissions that are not marked as final. This modification
uses a different label when Final is not appropriate. It also
clarifies whether the best or last submission is chosen for grading.

# Description

**What?**

See above. This is issue EDIT-768 in Jira.

**Why?**

As described in Jira

**How?**

[ANSWER HERE]

# Testing

**Remember to add or update unit tests for new features and changes.**

* How to [test your changes in A-plus](https://github.com/apluslms/a-plus/tree/master/doc#running-tests-and-updating-translations)
* How to [test accessibility](https://wiki.aalto.fi/display/EDIT/How+to+check+the+accessibility+of+pull+requests)

**What type of test did you run?**

- [ ] Accessibility test using the [WAVE](https://wave.webaim.org/extension/) extension.
- [ ] Django unit tests.
- [ ] Selenium tests.
- [ ] Other test. *(Add a description below)*
- [X] Manual testing.

Tested different combinations with few submissions by same user.

**Did you test the changes in**

- [X] Chrome
- [ ] Firefox
- [ ] This pull request cannot be tested in the browser.

**Think of what is affected by these changes and could become broken**

# Translation

- [X] Did you modify or add new strings in the user interface? ([Read about how to create translation](https://github.com/apluslms/a-plus/tree/master/doc#running-tests-and-updating-translations))

# Programming style

- [X] Did you follow our [style guides](https://apluslms.github.io/contribute/styleguides/)?

# Have you updated the README or other relevant documentation?

- [ ] documents inside the doc directory.
- [ ] README.md.
- [ ] Aplus Manual.
- [ ] Other documentation (mention below which documentation).

# Is it Done?

- [ ] Reviewer has finished the code review
- [ ] After the review, the developer has made changes accordingly
- [ ] Customer/Teacher has accepted the implementation of the feature

*Clean up your git commit history before submitting the pull request!*

### CODE REVIEW

File: exercise/templates/exercise/staff/inspect_submission.html [hunk 1]
```diff
@@ -81,6 +81,13 @@
 						</form>
 					{% endif %}
 				{% endif %}
+				{% if not submission.force_exercise_points %}
+					{% if exercise.grading_mode == 1 %}
+						{% translate "GRADING_MODE_BEST_EXPLAIN" %}
+					{% elif exercise.grading_mode == 2 %}
+						{% translate "GRADING_MODE_LAST_EXPLAIN" %}
+					{% endif %}
```
Comment (skulonen) [inline exercise/templates/exercise/staff/inspect_submission.html:89]: I don't think it's a good idea to hardcode enum values into the template, but I guess it's not possible to access the enum directly from the template. Could you generate the explanation string in the view code instead? Or maybe we could make a generic template tag that compares enum values by their string representation instead? Something like: `exercise.grading_mode|enum_equal:"BEST"`

File: locale/en/LC_MESSAGES/django.po [hunk 2]
```diff
@@ -3867,6 +3871,14 @@ msgstr "Assess next submitter manually"
...
 msgstr "Add one extra submission for the student in this exercise."
 
+#: exercise/templates/exercise/staff/inspect_submission.html
+msgid "GRADING_MODE_BEST_EXPLAIN"
+msgstr "Choosing highest grade"
+
+#: exercise/templates/exercise/staff/inspect_submission.html
+msgid "GRADING_MODE_LAST_EXPLAIN"
+msgstr "Choosing latest submission"
+
```
Comment (skulonen) [inline locale/en/LC_MESSAGES/django.po:3881]: Is it clear what "choosing" means? To me, "choosing" makes it sound like the user is expected to choose something.

File: exercise/templates/exercise/staff/_submissions_table_compact.html [hunk 3]
```diff
@@ -56,11 +56,19 @@
...
+						class="label label-primary"
+						data-toggle="tooltip"
+						title="{% translate 'SUBMISSION_FINAL_TOOLTIP' %}"
+					>{% translate 'FINAL' %}</span>
+					{% else %}
+					<span
+						class="glyphicon glyphicon-star"
+						data-toggle="tooltip"
+						title="{% translate 'SUBMISSION_CHOSEN_TOOLTIP' %}"
+					></span>
```
Comment (skulonen) [inline exercise/templates/exercise/staff/_submissions_table_compact.html:70]: This isn't a huge issue, but I think the gray star next to a colorful label looks a bit awkward. Could it perhaps be part of the points label or have its own colored label next to it?

**Comment by skulonen** [comment]
The logic seems valid to me. I wrote some nitpicks about things that I didn't like, but overall this looks good to me.

File: exercise/templates/exercise/staff/inspect_submission.html (hunk 1 above)
Comment (PasiSa) [reply exercise/templates/exercise/staff/inspect_submission.html]: True. Now choosing the text on the view side and just passing the string

File: locale/en/LC_MESSAGES/django.po (hunk 2 above)
Comment (PasiSa) [reply locale/en/LC_MESSAGES/django.po]: On a second thought I decided to use the existing translations "Grading mode" and "Best/Last Submission", which should be clear enough for the course staff. Anyway these texts were my own idea, and not a teacher request, so we could just leave it out as well.

File: exercise/templates/exercise/staff/_submissions_table_compact.html (hunk 3 above)
Comment (PasiSa) [reply exercise/templates/exercise/staff/_submissions_table_compact.html]: Now including the star inside a blue label. Any better?

File: locale/en/LC_MESSAGES/django.po (hunk 2 above)
Comment (skulonen) [reply locale/en/LC_MESSAGES/django.po]: Nice! I like this a lot.

File: exercise/templates/exercise/staff/inspect_submission.html (hunk 1 above)
Comment (skulonen) [reply exercise/templates/exercise/staff/inspect_submission.html]: Looks good!

File: exercise/templates/exercise/staff/_submissions_table_compact.html (hunk 3 above)
Comment (skulonen) [reply exercise/templates/exercise/staff/_submissions_table_compact.html]: To be honest I think this looks way worse :joy:
The star is off center and the label's height doesn't match the points label next to it. Plus it looks like the Somalian flag. I think we should go with your earlier version, it looked good enough.

**Review by Mankro — CHANGES_REQUESTED** [review]
This is good otherwise, but the way translated strings are concatenated does not support all languages. It is quick to change that, thus it could still be done before merging.

**Comment by PasiSa** [comment]
> This is good otherwise, but the way translated strings are concatenated does not support all languages. It is quick to change that, thus it could still be done before merging.

Done requested changes with one deviation: I used "Grading mode: {}" instead of proposed "{}: {}" which would have felt a bit awkward to me, despite the existing "Grading mode" model label. But then again, who knows, perhaps in some exotic language the model label text would not be usable as such in this situation anyway.

**Review by Mankro — APPROVED** [review]
Great, thanks!

**Pull request merged**